docs: homepage, concepts cleanup (#2643)
caos/zitadel: docs/docs/guides/architecture-scenarios/b2c.mdx

---
title: B2C
---

import Column from '../../../src/components/column';

> ## Business to Customer

Reviewer: ```suggestion
## Business to Consumer
```

This guide introduces you to the grouping and structuring of ZITADEL projects which forms the base for all projects. This can be used as a quick start to the [B2B scenario](./b2b), which is merely focused on planning considerations if you are having projects with multiple organizations.

The journey of this guide starts with creating an Organization, the outermost layer of ZITADEL, as it is the vessel for projects, roles, applications and users.
Creation can be done from [ZITADEL Console](https://console.zitadel.ch/org/create). You can choose your current account for the organization owner or create a new one.

Depending on your Software Development Life Cycle (SDLC) you will have to create multiple organizations to keep your data from different environments separate.

| Environment | Organization Name     | Description             |
| :---------- | :-------------------- | ----------------------- |
| Development | **acme-dev**          | The development environment and organization |
| Testing     | **acme-test**         | An environment for testing and production preview |

> | Production  | **acme-prod**         | The production environment and organization |

Reviewer: Do we really mention this as "best practice" scenario? In my opinion we recommended using different projects for different environments? also this would not match with our new pricing, because b2c only gets 2 organizations

Author: Then a dev user (fake account) could be used in production? would we want that? 🤔

Author: I would always use the outermost layer possible, which is an organization.

Author: @fforootd what are your thoughts?

Reviewer 2: @fgerschwiler 
>  also this would not match with our new pricing, because b2c only gets 2 organizations

In that case it might be a good nudge 😁 

@peintnermax 
> I would always use the outermost layer possible, which is an organization.
Well this is true to some degree, but for now i would say this is out of scope. Actually both ways can work and I think we should leave this open for customers choice 😄

### Custom domain

> Right after org creation you'll be greeted which the domain section of your organization. ZITADEL automatically creates a custom domain of the form `[orgname].zitadel.ch`, but you can set your own by saving a verification file on the specified location.  

Reviewer: ```suggestion
Right after org creation you'll be greeted with the domain section of your organization. ZITADEL automatically creates a custom domain of the form `[orgname].zitadel.ch`, but you can set your own by saving a verification file on the specified location.  
```

We recommend that you create your domains early as they create a sense of confidence and trust for your application and changes later on might create additional migration effort.
You can read more about how ZITADEL handles usernames [here](../basics/organizations#how-zitadel-handles-usernames).

### Data Provisioning

ZITADEL gives you a basic storage for users and manages phone and email addresses. It also allows you to store your own application data such as preferences or external identifiers to the metadata of a user.

If you are migrating an existing project and you already have an external identity store you can consider bulk importing your user datasets.
Read our [Management API definitions](../../apis/proto/management#importhumanuser) for more info. If the users email is not verified or no password is set, a initialization mail will be send. 

:::info
Requests to the management API are rate limited. Read our [Rate limit Policy](../../legal/rate-limit-policy) for more info.
:::

### User Authentication

> User Authentication can be performed in multiple ways. Default method in ZITADEL is username and password.

Reviewer 3: ```suggestion
User Authentication can be performed in multiple ways. Default method in ZITADEL is username and password with MFA enabled.
```
@fgerschwiler isn't MFA enabled by default?

When planning your application consider the following questions about User Authentication:

- What are the methods to authenticate your users?
- Where will your users enter their credentials?
- Do you offer password authentication?
- What do you do to keep your users credentials safe?
- Do you offer Multifactor Authentication?
- Do you offer Login via Identity Provider?
- Which languages do you have to provide?

When looking at this questions, you may have to admit that building an Identity Management System is much more complex and complicated than you thought initially and implementing if yourself may be too much work. 
Particularly because you should focus building your applications.

### Federation

ZITADEL supports signup with OIDC Identity providers as well as JWT Identity Providers. On signup, ZITADEL imports user information to the own profile.
//TODO extend this passage

### Hosted Login

ZITADEL offers a "secure by default approach" and comes with a Hosted Login which is a fixed endpoint for your users to authenticate. 
It's safe and secure and comes with Multifactor, Federated Login and Passwordless capabilities.

> OIDC (OpenID Connect) opens the doors Single Sign On (SSO). Especially if you have more than one application, you may want a central and familiar authentication experience. 

Reviewer: ```suggestion
OIDC (OpenID Connect) opens the doors to Single Sign On (SSO). Especially if you have more than one application, you may want a central and familiar authentication experience. 
```

With SSO, ZITADEL provides a seamless experience across all your apps.

### Branding

<Column>
<div>

Branding and customization is a very important part of your application. 
With ZITADEL you can modify colors, logos, icons as well as configure your typography styles, such that you can provide a consistent design throughout your applications.
In addition to visual modifications, you can edit notification texts for your users. 
ZITADEL gives you handlebar similar templating for variables. Of course you can define texts for any language.
We'd appreciate if you could contribute to our repo with translations of your language. Read on how to contribute [here](../../guides/customization/texts).

> Note that your console design changes to your design too

</div>
<img src="/img/guides/branding.jpeg" alt="branding in console"/>
</Column>

### Projects and applications

As our Hosted Login is a separate authentication screen, you have to determine how you are directing your users from your applications.

> ZITADELs Applications live under ZITADELs Projects. You may add multiple Applications for your different apps (Native, Web, User Agent, or API). When setting up you applications consider reading our guide about [Authentication Flows](../authentication/login-users).

Reviewer 3: ```suggestion
ZITADEL's Applications live under ZITADEL's Projects. You may add multiple applications for your different client-types (Native, Web, User Agent, or API). When setting up your applications consider reading our guide about [Authentication Flows](../authentication/login-users).
```

### Access Control

By having authenticated a user you need to ensure users and services have access to your application and APIs. This process is called Access Control and comprises User Authentication, Authorization and Policy Enforcement.
Take the following considerations:

- Does your application call your own APIs?
- Does your application need to call third-party APIs?
- Do **you** offer an API for third-party applications?

The data required to check if a user has access to a certain API is stored within a user grant. This information typically is stored within roles or custom claims and can be accessed with an `access` or OIDC `id` token.

> //TODO Link to AuhtZ guide

Reviewer: integrate link
